streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/bolt/CustomProcessorBolt.java

```diff
@@ -203,30 +170,35 @@ public void prepare (Map stormConf, TopologyContext context, OutputCollector col
     @Override
     protected void process (Tuple input, StreamlineEvent event) {
         try {
-            Map<String, Object> mappedEvent = new HashMap<>();
-            if (!inputSchemaMap.containsKey(input.getSourceStreamId()))
-                throw new RuntimeException("Received an event on an input stream that does not have mapping for input schema fields");
-            // Create a new mapped event based on mapping to pass it to CP implementation
-            for (Map.Entry<String, String> entry: inputSchemaMap.get(input.getSourceStreamId()).entrySet()) {
-                if (event.get(entry.getValue()) != null) {
-                    mappedEvent.put(entry.getKey(), event.get(entry.getValue()));
+            StreamlineEvent toProcess = null;
+            if (inputSchemaMap == null || inputSchemaMap.isEmpty() || !inputSchemaMap.containsKey(input.getSourceStreamId())) {
```

Comment:
1. Can it ever be empty or null? This can be a private final initialized to an empty map.
2. use get to retrieve the mapping based on the stream id before the `if` and check for null and reuse it in the else.

Reply:
Yes, it can be empty or null. In case of a CP registered with optional input schema, UI will not pass any mapping and hence it will be empty or null. In this case we pass the event received in the CustomProcessorBolt as is to the CP implementation. Your suggestion is better in general, but I think in this case it will make the code a bit complicated. If you are okay, we can leave it as is.